enterprise/server/backends/redis_metrics_collector/redis_metrics_collector.go

```diff
@@ -62,3 +62,14 @@ func (c *collector) ReadCounts(ctx context.Context, key string) (map[string]int6
 func (c *collector) Delete(ctx context.Context, key string) error {
 	return c.rdb.Del(ctx, key).Err()
 }
+
+func (c *collector) BufferDelay() time.Duration {
+	// Buffer is operating in synchronous mode; don't report a buffer delay.
+	if *redisutil.CommandBufferFlushPeriod == 0 {
+		return 0
+	}
+	// Report the buffer delay as slightly more than the command buffer flush
+	// period, to allow some time to actually execute the Redis commands, plus
+	// a little extra to account for clock skew across apps.
+	return *redisutil.CommandBufferFlushPeriod + 250*time.Millisecond
```

Comment:
rather than "polluting" your implementation with this logic to make the build_event_handler code work -- can you add the delay there at that callsite and return the true value here?

Reply:
I think I see what you mean -- with a name like `BufferDelay`, this ought to reflect the actual amount of time that we're buffering.

My intent was actually to reflect the time that a caller has to wait between calling `IncrementCounts()` and the time that the counts can be "read back" via `ReadCounts()`. So I renamed this to `ReadbackDelay()`, and since this delay depends both on the buffering delay as well as the storage mechanism used, I think it makes sense for the 250ms value to be an implementation detail here. The 250ms is a worst case upperbound that is specific to Redis -- other implementations could have higher or lower values here.

Side note, I removed the comment about clock skew, since that is irrelevant as we're dealing with relative durations and not absolute timestamps

Reply:
Resolved offline to just make these constants flags, instead of adding an interface